diff -urN nethack-3.4.3/arkenstone.readme nethack-3.4.3-arkenstone/arkenstone.readme
--- nethack-3.4.3/arkenstone.readme	1969-12-31 17:00:00.000000000 -0700
+++ nethack-3.4.3-arkenstone/arkenstone.readme	2006-02-25 20:15:46.000000000 -0700
@@ -0,0 +1,43 @@
+Arkenstone patch for Nethack 3.4.3
+
+Recently, in a thread in rec.games.roguelike.nethack, Jove suggested that
+dwarf kings could be occasionally generated carrying the Arkenstone.  Later,
+Bard made a couple suggestions for what the Arkenstone would do.  I thought
+it sounded fun, so I went ahead and made a quick patch.
+
+Spoilerific description:
+
+The Arkenstone is a lawful artifact that grants conflict and aggravate
+monster when carried.  
+
+Dwarf kings will (very occasionally) be generated carrying it.
+
+If you are carrying it, demon lords will demand it instead of a monetary
+bribe.
+
+Breakdown of files changed:
+
+dat/data.base
+  Add a quote for the database entry for the Arkenstone.  Also use this
+  quote for a generic "gem," since there isn't one already in the database.
+  
+include/artifact.h
+  Add flags for SPFX_CONFL (conflict) and SPFX_AGGRM (aggravate monster)
+  
+include/artilist.h
+  Add artifact definition for the Arkenstone
+
+include/config.h
+  Add definition of ARKENSTONE
+
+src/artifact.c
+  Add cases for SPFX_CONFL and SPFX_AGGRM in set_artifact_intrinsic()
+
+src/makemon.c
+  Give the Arkenstone occasionally to dwarf kings
+
+src/minion.c
+  Make demons demand the Arkenstone before a monetary bribe
+
+src/objnam.c
+  Modify the() to make sure that Arkenstone always has "the" before it
diff -urN nethack-3.4.3/dat/data.base nethack-3.4.3-arkenstone/dat/data.base
--- nethack-3.4.3/dat/data.base	2004-12-05 19:49:38.000000000 -0700
+++ nethack-3.4.3-arkenstone/dat/data.base	2006-02-25 20:15:46.000000000 -0700
@@ -201,6 +201,23 @@
 	the Swords, Lord of the Seven Darks, Lord of the Higher Hell
 	and many more names besides.
 		[ Elric of Melnibone, by Michael Moorcock ]
+*arkenstone
+gem
+	It was the Arkenstone, the Heart of the Mountain. So Bilbo
+	guessed from Thorin's description; but indeed there could not
+	be two such gems, even in so marvellous a hoard, even in all
+	the world.  Ever as he climbed, the same white gleam had shone
+	before him and drawn his feet towards it.  Slowly it grew to a
+	little globe of pallid light.  Now as he came near, it was
+	tinged with a flickering sparkle of many colours at the
+	surface, reflected and splintered from the wavering light of
+	his torch.  At last he looked down upon it, and he caught his
+	breath. The great jewel shone before his feet of its own inner
+	light, and yet, cut and fashioned by the dwarves, who had dug
+	it from the heart of the mountain long ago, it  took all light
+	that fell upon it and changed it into ten thousand sparks of
+	white radiance shot with glints of the rainbow.
+		[ The Hobbit, by J.R.R. Tolkien ]
 *arrow
 	I shot an arrow into the air,
 	It fell to earth, I knew not where;
diff -urN nethack-3.4.3/include/artifact.h nethack-3.4.3-arkenstone/include/artifact.h
--- nethack-3.4.3/include/artifact.h	2004-12-05 19:49:38.000000000 -0700
+++ nethack-3.4.3-arkenstone/include/artifact.h	2006-02-25 20:15:46.000000000 -0700
@@ -34,6 +34,10 @@
 #define SPFX_DBONUS 0x1F00000L	/* attack bonus mask */
 #define SPFX_XRAY   0x2000000L	/* gives X-RAY vision to player */
 #define SPFX_REFLECT 0x4000000L /* Reflection */
+#ifdef ARKENSTONE
+#define SPFX_CONFL  0x8000000L  /* Conflict */
+#define SPFX_AGGRM  0x10000000L /* Aggravate monster */
+#endif
 
 
 struct artifact {
diff -urN nethack-3.4.3/include/artilist.h nethack-3.4.3-arkenstone/include/artilist.h
--- nethack-3.4.3/include/artilist.h	2004-12-05 19:49:38.000000000 -0700
+++ nethack-3.4.3-arkenstone/include/artilist.h	2006-02-25 20:15:46.000000000 -0700
@@ -148,6 +148,12 @@
 	(SPFX_RESTR|SPFX_DFLAG2), 0, M2_UNDEAD,
 	PHYS(5,0),	DFNS(AD_BLND),	NO_CARY,	0, A_LAWFUL, NON_PM, NON_PM, 1500L ),
 
+#ifdef ARKENSTONE
+A("The Arkenstone", DIAMOND,
+        SPFX_RESTR, (SPFX_CONFL|SPFX_AGGRM), 0,
+        NO_ATTK, NO_DFNS, NO_CARY, 0, A_LAWFUL, NON_PM, NON_PM, 8000L ),
+#endif
+
 /*
  *	The artifacts for the quest dungeon, all self-willed.
  */
diff -urN nethack-3.4.3/include/config.h nethack-3.4.3-arkenstone/include/config.h
--- nethack-3.4.3/include/config.h	2004-12-05 19:49:38.000000000 -0700
+++ nethack-3.4.3-arkenstone/include/config.h	2006-02-25 20:15:46.000000000 -0700
@@ -340,6 +340,8 @@
 #define EXP_ON_BOTL	/* Show experience on bottom line */
 /* #define SCORE_ON_BOTL */	/* added by Gary Erickson (erickson@ucivax) */
 
+#define ARKENSTONE /* Add "the Arkenstone," an artifact. */
+
 /*
  * Section 5:  EXPERIMENTAL STUFF
  *
diff -urN nethack-3.4.3/src/artifact.c nethack-3.4.3-arkenstone/src/artifact.c
--- nethack-3.4.3/src/artifact.c	2004-12-05 19:49:38.000000000 -0700
+++ nethack-3.4.3-arkenstone/src/artifact.c	2006-02-25 20:15:46.000000000 -0700
@@ -476,7 +476,16 @@
 	    if (on) EReflecting |= wp_mask;
 	    else EReflecting &= ~wp_mask;
 	}
-
+#ifdef ARKENSTONE
+        if (spfx & SPFX_CONFL) {
+            if (on) EConflict |= wp_mask;
+            else EConflict &= ~wp_mask;
+        }
+        if (spfx & SPFX_AGGRM) {
+            if (on) EAggravate_monster |= wp_mask;
+            else EAggravate_monster &= ~wp_mask;
+        }
+#endif /* ARKENSTONE */
 	if(wp_mask == W_ART && !on && oart->inv_prop) {
 	    /* might have to turn off invoked power too */
 	    if (oart->inv_prop <= LAST_PROP &&
diff -urN nethack-3.4.3/src/makemon.c nethack-3.4.3-arkenstone/src/makemon.c
--- nethack-3.4.3/src/makemon.c	2004-12-05 19:49:38.000000000 -0700
+++ nethack-3.4.3-arkenstone/src/makemon.c	2006-02-25 20:15:46.000000000 -0700
@@ -316,6 +316,15 @@
 		    } else {
 			(void)mongets(mtmp, !rn2(3) ? PICK_AXE : DAGGER);
 		    }
+#ifdef ARKENSTONE
+                    if((mm == PM_DWARF_KING) && !rn2(75) &&
+                        !exist_artifact(DIAMOND, artiname(ART_ARKENSTONE))) {
+                        otmp = mksobj(DIAMOND, FALSE, FALSE);
+                        otmp->quan = 1L;
+                        otmp = oname(otmp, artiname(ART_ARKENSTONE));
+                        (void) mpickobj(mtmp, otmp);
+                    }
+#endif /* ARKENSTONE */
 		}
 		break;
 # ifdef KOPS
diff -urN nethack-3.4.3/src/minion.c nethack-3.4.3-arkenstone/src/minion.c
--- nethack-3.4.3/src/minion.c	2004-12-05 19:49:38.000000000 -0700
+++ nethack-3.4.3-arkenstone/src/minion.c	2006-02-25 20:15:46.000000000 -0700
@@ -142,6 +142,11 @@
 demon_talk(mtmp)		/* returns 1 if it won't attack. */
 register struct monst *mtmp;
 {
+#ifdef ARKENSTONE
+        struct obj *otmp;
+        char qbuf[QBUFSZ];
+#endif
+
 	long cash, demand, offer;
 
 	if (uwep && uwep->oartifact == ART_EXCALIBUR) {
@@ -172,6 +177,40 @@
 	demand = (cash * (rnd(80) + 20 * Athome)) /
 	    (100 * (1 + (sgn(u.ualign.type) == sgn(mtmp->data->maligntyp))));
 
+#ifdef ARKENSTONE
+        /* If you have the Arkenstone, the demon demands 
+         * it instead of gold. */
+        for(otmp = invent; otmp; otmp = otmp->nobj) {
+            if(otmp->oartifact == ART_ARKENSTONE) break;
+        }
+
+        if(otmp) {
+            /* You have the Arkenstone; demand it unless the monster has the
+             * Amulet (same justification as below.)  If it does, then we
+             * will drop into the normal monetary bribe code (which can't
+             * be satisfied.) */
+            
+            if(!mon_has_amulet(mtmp)) {
+                pline("%s demands %s for safe passage.",
+                    Amonnam(mtmp), the(xname(otmp)));
+                Sprintf(qbuf, "Give it to %s?", mhim(mtmp));
+                if(yn(qbuf) == 'y') {
+                    pline("%s vanishes, laughing about cowardly mortals.",
+                          Amonnam(mtmp));
+                    freeinv(otmp);
+                    (void) mpickobj(mtmp, otmp);
+                    mongone(mtmp);
+                    return 1;
+                } else {
+                    pline("%s gets angry...", Amonnam(mtmp));
+                    mtmp->mpeaceful = 0;
+                    set_malign(mtmp);
+                    return 0;
+                }
+            }
+        }
+#endif
+
 	if (!demand) {		/* you have no gold */
 	    mtmp->mpeaceful = 0;
 	    set_malign(mtmp);
diff -urN nethack-3.4.3/src/objnam.c nethack-3.4.3-arkenstone/src/objnam.c
--- nethack-3.4.3/src/objnam.c	2004-12-05 19:49:38.000000000 -0700
+++ nethack-3.4.3-arkenstone/src/objnam.c	2006-02-25 20:15:46.000000000 -0700
@@ -993,7 +993,15 @@
 		else if (!named && (l = strlen(str)) >= 31 &&
 		      !strcmp(&str[l - 31], "Platinum Yendorian Express Card"))
 		    insert_the = TRUE;
-	    }
+	    } 
+#ifdef ARKENSTONE            
+            /* another stupid case for "the Arkenstone" */
+            else if(!strstri(str, " named ") && 
+                    (l = strlen(str)) >= 10 && 
+                    !strcmp(&str[l - 10], "Arkenstone")) {
+                insert_the = TRUE;
+            }
+#endif
 	}
 	if (insert_the)
 	    Strcpy(buf, "the ");
